### PR TITLE
Add basic blocks to Stonemason for DO

### LINKED ISSUE #9282: [BUG] Stonemason wont accept just stone in his "Custom Recipes"
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Verison?

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### What were you playing at the time? Were you able to reproduce it in both settings?

- [ ] Single Player
- [X] Multi Player

### Minecraft Version

1.20.1

### MineColonies Version

1.20.1-1.1.28.-RELEASE

### Structurize Version

1.20-1.0.584-BETA

### Related Mods and their Versions

1. Forge version: 47.1.3
2. All the mods 9 Version: 0.0.45

### Current Behavior

Stonemason wont accept just stone in his "Custom Recipes"
As in the architects cutters recipe for for example: Stone Panels.  

He has the space.

### Expected Behavior

Stonemason will accept just stone in his "Custom Recipes"

### Reproduction Steps

In my case its Cavern Stonemason lvl 2.
image
Press "Teach" and the error on the bottom left shows up.
He has accepted other recipes before and after this.
image
image

I am Running All the mods 9 0.0.45.
I know AE2, which is installed, caused an issue before, so maybe thats it.

### Logs

https://pastebin.com/0vp7z7Wy

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by Mekle001**
Closes #9282 

# Changes proposed in this pull request:
-Add a small number of blocks that are always accepted for DO ingredients to Stonemason

This change compensates for the fact that Stone, Netherrack, etc all get turned into storageblocks by Gregtech

Review please

### CODE REVIEW

**Review by Raycoms — DISMISSED** [review]

File: src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingStonemason.java [hunk 1]
```diff
@@ -96,7 +100,18 @@ public DOCraftingModule(final JobEntry jobEntry)
          */
         public @NotNull static OptionalPredicate<ItemStack> getStaticIngredientValidator()
         {
-            return CraftingUtils.getIngredientValidatorBasedOnTags(CRAFTING_STONEMASON);
+
+            OptionalPredicate<ItemStack> verifyStone = stack -> 
+            {
+                if(stack.is(Items.STONE) || stack.is(Items.NETHERRACK) 
+                || stack.is(Blocks.PRISMARINE.asItem()) || stack.is(Blocks.DARK_PRISMARINE.asItem()))
```
Comment (uecasm) [inline src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingStonemason.java]: This whitespacing irks me.

Also, perhaps worth introducing a new tag for this?

Comment (Mekle001) [reply src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingStonemason.java]: I've thought about a new tag, and I may still do that, but that is a much bigger change, and I'm not sure we'll ever need to do this again.

Comment (uecasm) [reply src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingStonemason.java]: It'd be an extension point that lets modpacks add more stonemason-specific blocks if they have other things that cause weirdness.  (e.g. if they add even more stone-ish "storage blocks", or something else.)

And other than picking a name, it's very easy to add a new tag.  Just a few lines of code and a `runData`.

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingStonemason.java]: I agree, a tag would be easy, and make extending things easier on the long term

Comment (Mekle001) [reply src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingStonemason.java]: Ok. Switched to a tag

**Review by uecasm — APPROVED** [review]

**Pull request merged**